### ISSUE TITLE
`rubocop --auto-gen-config` cannot exclude Style/HashSyntax on Ruby 3.1

### ISSUE THREAD

**Issue by r7kamura** [rubocop/rubocop#10639]
When I run `rubocop` to the following code with TargetRubyVersion 3.1, an offense to Style/HashSyntax is detected:

```ruby
# frozen_string_literal: true

{ a: a }
```

After that, I run `rubocop --auto-gen-config`, and the following .rubocop_todo.yml is generated:

```yaml
# This configuration was generated by
# `rubocop --auto-gen-config`
# on 2022-05-15 23:30:32 UTC using RuboCop version 1.29.1.
# The point is for the user to remove these configuration records
# one by one as the offenses are removed from the code base.
# Note that changes in the inspected code, or installation of new
# versions of RuboCop, may require this file to be generated again.

# Offense count: 1
# This cop supports safe auto-correction (--auto-correct).
# Configuration parameters: EnforcedShorthandSyntax, UseHashRocketsWithSymbolValues, PreferHashRocketsForNonAlnumEndingSymbols.
# SupportedStyles: ruby19, hash_rockets, no_mixed_keys, ruby19_no_mixed_keys
# SupportedShorthandSyntax: always, never, either
Style/HashSyntax:
  EnforcedStyle: ruby19
```

and then re-reun `rubocop`, but the offense detected earlier will continue to be detected.

## Expected behavior

I think the following .rubocop_todo.yml should be generated:

```yaml
Style/HashSyntax:
  EnforcedShorthandSyntax: never
```

or

```yaml
Style/HashSyntax:
  Exclude:
    - '/path/to/foo.rb'
```

## Actual behavior

As explained at the beginning, the following .rubocop_todo.yml is generated.

```yaml
Style/HashSyntax:
  EnforcedStyle: ruby19
```

## Steps to reproduce the problem

Create some files...

```yaml
# .rubocop.yml
AllCops:
  NewCops: enable
  TargetRubyVersion: 3.1
```

```ruby
# frozen_string_literal: true

{ a: a }
```

then run `rubocop --auto-gen-config`:

```console
$ rubocop --auto-gen-config
Added inheritance from `.rubocop_todo.yml` in `.rubocop.yml`.
Phase 1 of 2: run Layout/LineLength cop
Inspecting 1 file
.

1 file inspected, no offenses detected
Created .rubocop_todo.yml.
Phase 2 of 2: run all cops
Inspecting 1 file
C

1 file inspected, 1 offense detected, 1 offense auto-correctable
Created .rubocop_todo.yml.
```

and then run `rubocop`:

```console
$ rubocop 
Inspecting 1 file
C

Offenses:

example.rb:3:6: C: [Correctable] Style/HashSyntax: Omit the hash value.
{ a: a }
     ^

1 file inspected, 1 offense detected, 1 offense auto-correctable
```

## RuboCop version

```
$ rubocop -V
1.29.1 (using Parser 3.1.2.0, rubocop-ast 1.18.0, running on ruby 3.1.2 x86_64-linux)
```
